webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TraitTestApiFromYamlFiles.scala

```diff
@@ -290,72 +276,118 @@ trait TraitTestApiFromYamlFiles extends Specification with BoxSpecMatcher with J
+                case (eb: EmptyBox, i) =>
+                  val f = eb ?~! s"I wasn't able to run the ${i}th tests in file ${name}"
+                  ApplicationLogger.error(f.messageChain)
+
+                  // erg. There must be a better way
+                  zio.test.test(
+                    s"[${i}] failing test ${i}: can not read description. The yaml description ${i} in ${name} cannot be read: ${f.messageChain}"
+                  )(
+                    assert(true)(Assertion.isFalse)
+                  )
+
+                case (Full(test), i) =>
+                  zio.test.test(s"[${i}] ${test.description}") {
+                    val mockReq = new MockHttpServletRequest("http://localhost:8080")
+                    mockReq.method = test.method
+                    val p       = test.url.split('?')
+                    mockReq.path = p.head // exists, split always returns at least one element
+                    mockReq.queryString = if (p.size > 1) p(1) else ""
+                    mockReq.body = test.queryBody.getBytes(StandardCharsets.UTF_8)
+                    mockReq.headers = test.headers.map { h =>
+                      val parts = h.split(":")
+                      (parts(0), List(parts.tail.mkString(":").trim))
+                    }.toMap
+                    // query string may have already set some params
+                    mockReq.parameters = mockReq.parameters ++ test.params
+                    mockReq.contentType = mockReq.headers.get("Content-Type").flatMap(_.headOption).getOrElse("text/plain")
+
+                    restTest.execRequestResponseZioTest(mockReq) { response =>
+                      (for {
+                        clean        <- response.map(cleanResponse).toPureResult.left.map(_.fullMsg)
+                        expectedJson <- cleanJson(test.responseContent)
+                        responseJson <- cleanJson(clean._2)
+                      } yield (clean._1, expectedJson, responseJson)) match {
+                        case Left(s)                         =>
+                          assert(s)(Assertion(TestArrow.make[String, Boolean](s => TestTrace.fail(s))))
+                        case Right((code, expJson, resJons)) =>
+                          assertTrue(
+                            code == test.responseCode,
+                            resJons == expJson
+                          )
+                      }
+                    }
+                  }
+
+              }).succeed
+            })
           }
+          tests
+        } @@ TestAspect.sequential).succeed
+    }
+  }
+
+  def cleanJson(json: String): Either[String, String] = {
+    import zio.json.*
+    import zio.json.ast.*
+    json.fromJson[Json].map(_.toJsonPretty) match {
+      case Left(err)   =>
+        // here, we need to take care of the case of a pure string. Rule of thumb: if the first char is not json-ok, fallback
+        err match {
+          case "(unexpected 'O')" => Right(json)
```

Comment:
in that case, we assume it's not json, but just a simple string. Return it as is.